code/modules/mob/mob_helpers.dm

```diff
@@ -22,6 +22,42 @@
 			return 0
 	return 1
 
+/mob/proc/get_screen_colour()
+
+/mob/proc/update_client_colour(var/time = 10) //Update the mob's client.color with an animation the specified time in length.
+	if(!client) //No client_colour without client. If the player logs back in they'll be back through here anyway.
+		return
+	client.colour_transition(get_screen_colour(), time = time) //Get the colour matrix we're going to transition to depending on relevance (magic glasses first, eyes second).
+
+/mob/living/carbon/human/update_client_colour(var/time = 10) /*Handle the activation/deactivation of colourblindness on the mob.
+	This was designed so mobs with 'incurable colourblindness' (e.g. Farwas/Wolpins, which is not genetics-based), won't interfere with greater-form mobs during transformations.
+	The intended behaviour is that, when transforming into a lesser form (monkey) as a greater form (human) with colourblindness,
+	you will keep it even when you transform back into a human, although genetics can still manipulate it (turn off/on) as normal.
+	Works hand-in-hand with eye insertion/removal procs in code\modules\surgery\organs\organ_internal.dm. See that file for the rest of how this is done.*/
+
+	var/obj/item/organ/internal/eyes/eyes = get_int_organ(/obj/item/organ/internal/eyes)
+	if(eyes && !eyes.robotic) //To avoid overriding the properties of cybernetic/mechassisted eyes, apply the colour matrix and darksight modifier only if the organ is organic.
+		var/list/cblind_properties = (eyes.colourblind_special) ? eyes.colourblind_special : null //If the mob's eyes have their own colourblindness parameters, gather them.
+		if(disabilities & COLOURBLIND) //If the mob's got the colourblind disability, apply the appropriate parameters to the eyes' colourmatrix and dark_view.
+			eyes.colourmatrix = (cblind_properties && cblind_properties["colour_matrix"]) ? cblind_properties["colour_matrix"] : MATRIX_GREYSCALE //If the mob's eyes have their own colourblindness colour matrix parameter, use it.
+			eyes.dark_view = (cblind_properties && cblind_properties["darkview"]) ? cblind_properties["darkview"] : eyes.dark_view //If the mob's eyes have their own colourblindness dark_view parameter, use it.
+		else if(eyes.colourmatrix && cblind_properties["darkview"] != initial(eyes.dark_view)) /*If the mob doesn't have the colourblind disability, correctly set the eyes' colourmatrix and dark_view defaults but only if it's meant to be curable..
+ 																								 These conditions are a /CRITICAL PART/ of what allows you to switch from (i.e.) colourblind Vulpkanin to Wolpin without suddenly finding yourself seeing colour.*/
+			eyes.colourmatrix = null //We're fine to null this out here because the mob doesn't have non genetics-based colourblindness or the colourblindness disability.

```

Comment:
Instead of having all this code to modify the `colourmatrix` and `dark_view` vars, why not make the code that accesses them simply pick the colorblind versions if needed? Preferably using new helper procs, so you can keep the checks in one place.

Also, this proc has a *ton* of comments. If you have to put a comment after almost every line, it either means the code itself needs to be more readable on its own, or you're adding comments to things you really shouldn't need to. Too many comments is almost as bad as not having enough.